payjoin/src/receive/v1/mod.rs

```diff
@@ -1070,4 +1088,128 @@ pub(crate) mod test {
         assert_eq!(original2, vec![1, 5, 4, 2, 6, 3]);
         assert_eq!(original3, vec![4, 5, 1, 2, 6, 3]);
     }
+
+    /// Add non-consensus data to inputs and outputs and verify it is excluded from the final PSBT
+    #[test]
+    fn test_prepare_psbt_excludes_non_consensus_fields() {
+        let proposal = unchecked_proposal_from_test_vector();
+        let mut processed_psbt = proposal.psbt.clone();
+
+        let secp = Secp256k1::new();
+
+        let sk = SecretKey::from_slice(&[1u8; 32]).expect("32-byte secret key");
```

Comment:
Can we use the dummy key pair method here as well.

Reply:
instead i just generated the one key at the top of the test and reused it